### PR TITLE
Change endpoint for parcel points

### PR DESCRIPTION
**Pull request by yekovalenkoa**
BC-break:
---
- Drop support for PHP < 8.2
- Drop support for PHPunit < 10
- MabeEnum has been replaced with PHP native Enum.

AC
---
- The endpoint from which parcel pickup points are retrieved has been changed.
- The `ParcelShop` DTO has been extended with the `features` field, which returns values from `FeatureEnum`.
- The `ParcelShop` DTO `openings` field has been expanded with detailed opening hours and breaks in the operation of the pickup point. Each day is now an `Openings` DTO.

### CODE REVIEW

File: src/Client/RequestTransformer.php [hunk 1]
```diff
@@ -27,10 +27,6 @@ public function transform(RequestInterface $request): HttpRequest
     {
         $uri = $this->configuration->getUrl() . $request->getEndpoint();
 
-        if (null !== $request->getUrlQuery()) {
-            $uri .= '?' . $request->getUrlQuery();
-        }
```
Comment (lukasz-falda) [inline src/Client/RequestTransformer.php:32]: why this is removed?

File: src/ConfigProvider.php [hunk 2]
```diff
@@ -4,38 +4,31 @@
...
     public function getUrl(): string
     {
-        $url = self::URL_MAP[$this->getCountryCode()->getValue()] ?? null;
-        if (null === $url) {
-            throw new \InvalidArgumentException('No url for provided country code.');
-        }
-
-        return $url;
+        return match ($this->getCountryCode()) {
+            CountryCodeEnum::Romania,
```
Comment (lukasz-falda) [inline src/ConfigProvider.php]: Why romania has changed url from `https://online.gls-romania.com/`

File: src/ConfigProvider.php [hunk 3]
```diff
@@ -4,38 +4,31 @@
...
+            CountryCodeEnum::Hungary => 'https://map.gls-hungary.com/',
+            CountryCodeEnum::Slovakia => 'https://map.gls-slovakia.com/',
+            CountryCodeEnum::Czech => 'https://map.gls-czech.com/',
+            CountryCodeEnum::Slovenia => 'https://map.gls-slovenia.com/',
+            CountryCodeEnum::Croatia => 'https://map.gls-croatia.com/',
+        };
```
Comment (lukasz-falda) [inline src/ConfigProvider.php]: Why this is better?
I preferred const for this map

File: src/Response/DTO/Address.php [hunk 4]
```diff
@@ -6,68 +6,29 @@
 
 class Address
```
Comment (lukasz-falda) [inline src/Response/DTO/Address.php]: maybe you can make these classes readonly

**Review by lukasz-falda — COMMENTED** [review]
1. It would be nice to drop supports in other PR and after that change endpoint.

2. Does the README file need updating?

File: src/ConfigProvider.php (hunk 2 above)
Comment (yekovalenkoa) [reply src/ConfigProvider.php]: `https://online.gls-romania.com/` is an old endpoint and is not supported by the current changes in this PR.
Romania does not have its own new endpoint, similar to other countries. Instead, we can use the following endpoint to retrieve the list of parcel pickup points: 
`https://map.gls-hungary.com/data/deliveryPoints/ro.json`

File: src/ConfigProvider.php (hunk 3 above)
Comment (yekovalenkoa) [reply src/ConfigProvider.php]: I fixed it

File: src/Response/DTO/Address.php (hunk 4 above)
Comment (yekovalenkoa) [reply src/Response/DTO/Address.php]: Sure, I fixed it.

**Comment by yekovalenkoa** [comment]
> 1. It would be nice to drop supports in other PR and after that change endpoint.

I agree, but a lot of changes would need to be made in the subsequent PRs.
So, I prefer one PR with 30 modified files rather than 2 or 3 PRs with 15-20 files changed in each.

> 2. Does the README file need updating?

No, an update is not needed.

**Review by lukasz-falda — APPROVED** [review]

File: src/Logger/GlsLogger.php [hunk 5]
```diff
@@ -89,7 +89,16 @@ public function clearRequestId(): void
 
     private function decorateMessage(string $message, ?UriInterface $uri): string
     {
-        return self::MESSAGE_PREFIX . $message . ' - ' . (null !== $uri ? $uri->getPath() . '?' . $uri->getQuery() : '');
+        $url = null;
+        if (null !== $uri) {
+            $url = $uri->getPath();
+
+            if ($uri->getQuery()) {
+                $url .= '?' . $uri->getQuery();
```
Comment (ropczan) [inline src/Logger/GlsLogger.php]: is it still needed?

File: src/Request/GetParcelShops.php [hunk 6]
```diff
@@ -4,68 +4,36 @@
...
-                'pclshopin' => null === $this->pclShopIn ? null : (int) $this->pclShopIn,
-                'parcellockin' => null === $this->parcelLockIn ? null : (int) $this->parcelLockIn,
-                'codhandler' => null === $this->codHandler ? null : (int) $this->codHandler,
-                'dropoff' => null === $this->dropOff ? null : (int) $this->dropOff,
-                'pclshopid' => $this->shopId,
-            ]
-        );
-
-        return empty($urlQuery) ? null : $urlQuery;
+        return match ($this->countryCode) {
```
Comment (ropczan) [inline src/Request/GetParcelShops.php]: seems to me better to do: `sprintf('%s.json',strtolower($this->countryCode->getValue()))`

if we need to handle another country then there is no need to change anything here

File: src/Response/DTO/ParcelShop.php [hunk 7]
```diff
@@ -4,178 +4,61 @@
...
-     */
-    public static function fromRawParcelShop(RawParcelShop $rawParcelShop, array $openings): self
+    public static function fromRawParcelShop(RawParcelShop $rawParcelShop): self
     {
-        $coordinates = null;
-
-        if (null !== $rawParcelShop->geolat && null !== $rawParcelShop->geolng) {
-            $coordinates = new Coordinates((float) $rawParcelShop->geolat, (float) $rawParcelShop->geolng);
-        }
+        $isCodHandler = in_array(FeatureEnum::AcceptsCash->value, $rawParcelShop->features, true);
```
Comment (ropczan) [inline src/Response/DTO/ParcelShop.php]: These variables are not reused, in my opinion they do not make order here at all just more confusion. I propose to remove the variables.

Comment (yekovalenkoa) [reply src/Response/DTO/ParcelShop.php]: Not a problem, I shortened the list of variables.

File: src/Request/GetParcelShops.php (hunk 6 above)
Comment (yekovalenkoa) [reply src/Request/GetParcelShops.php]: Done

File: src/Logger/GlsLogger.php (hunk 5 above)
Comment (yekovalenkoa) [reply src/Logger/GlsLogger.php]: Actually, this part is not really necessary. I removed it.

File: src/Response/DTO/ParcelShop.php [hunk 8]
```diff
@@ -31,16 +31,11 @@ public function __construct(
 
     public static function fromRawParcelShop(RawParcelShop $rawParcelShop): self
     {
-        $isCodHandler = in_array(FeatureEnum::AcceptsCash->value, $rawParcelShop->features, true);
-        $canPayByBankCard = in_array(FeatureEnum::AcceptsCard->value, $rawParcelShop->features, true);
-        $isParcelLocker = $rawParcelShop->type === ParcelShopTypeEnum::ParcelLocker->value;
         $contact = Address::fromResponse($rawParcelShop->contact);
```
Comment (ropczan) [inline src/Response/DTO/ParcelShop.php:34]: this one more could be removed

**Review by ropczan — APPROVED** [review]

File: src/Response/DTO/ParcelShop.php (hunk 8 above)
Comment (yekovalenkoa) [reply src/Response/DTO/ParcelShop.php:34]: This one is used in two places: [line 47](https://github.com/answear/gls-bundle/pull/7/files#diff-8e5a7e664b841584c3a4b31df4553b4c63d861cf0060a42f3fe0b9b96a94da27R47) and [line 53](https://github.com/answear/gls-bundle/pull/7/files#diff-8e5a7e664b841584c3a4b31df4553b4c63d861cf0060a42f3fe0b9b96a94da27R53)

**Pull request merged**